### PR TITLE
Adding `dimcli` from PyPI

### PR DESCRIPTION
**Pull request by sugatoray**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged. `NA`
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sugatoray** [comment]
## Note

The PyPI package of `dimcli` does not have `requirements.txt`, which is why as a workaround, I am currently copying it from `dimcli.egg-info/requires.txt`. The PR for this issue (https://github.com/digital-science/dimcli/issues/75) should fix that. Once it is fixed, removed the extra commands other than `{{ PYTHON }} -m pip install . -vv`

## Copying the `requirements.txt` file

```yaml
script:
    {% set reqfile = "requirements.txt" %}
    {% set reqsource = "dimcli.egg-info/requires.txt" %}
    # IF {{ reqfile }} is NOT PRESENT, COPY {{ reqfile }} from {{ reqsource }}
    - if [ ! -f {{ reqfile }} ] && [ -f {{ reqsource }} ]; then cp {{ reqsource }} {{ reqfile }}; fi  # [not win]
    ## - echo "IF NOT EXIST {{ reqfile }} ( IF EXIST {{ reqsource }} COPY {{ reqsource }} {{ reqfile }} ) )" | cmd.exe  # [win]
    ## - IF (-NOT(TEST-PATH -PATH {{ reqfile }})) { IF (TEST-PATH -PATH {{ reqsource }}) { COPY {{ reqsource }} {{ reqfile }} } }  # [win]
    - COPY {{ reqsource }} {{ reqfile }}  # [win]
    - {{ PYTHON }} -m pip install . -vv
```

**Comment by sugatoray** [comment]
Ready for review.

cc: @conda-forge/help-python

File: recipes/dimcli/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  entry_points:
+    - dimcli = dimcli.main_cli:main_cli
+    - dimcli_quicktest = dimcli.tests.quicktest:main
+  script:
+    {% set reqfile = "requirements.txt" %}
+    {% set reqsource = "dimcli.egg-info/requires.txt" %}
+    # IF {{ reqfile }} is NOT PRESENT, COPY {{ reqfile }} from {{ reqsource }}
+    - if [ ! -f {{ reqfile }} ] && [ -f {{ reqsource }} ]; then cp {{ reqsource }} {{ reqfile }}; fi  # [not win]
+    - copy dimcli.egg-info\requires.txt {{ reqfile }}  # [win]
+    - {{ PYTHON }} -m pip install . -vv
```
Comment (beckermr) [inline recipes/dimcli/meta.yaml:25]: Kudos for this bit of magic!

Comment (sugatoray) [reply recipes/dimcli/meta.yaml:25]: Hehe! Not quite!

If you see carefully, I was not able to apply the "IF" conditionals for windows. Any help/suggestion on what I should have done?

Comment (beckermr) [reply recipes/dimcli/meta.yaml:25]: Windows has its own syntax that I usually find by looking through other recipes. IDK If you tried that. I don't see the problem with always copying requires though.

**Comment by sugatoray** [comment]
Thank you, for the lightning fast merge! :zap: @beckermr

**Pull request merged**